pkg/storage/ec/client.go

```diff
@@ -28,45 +27,37 @@ var mon = monkit.Package()
 // Client defines an interface for storing erasure coded data to piece store nodes
 type Client interface {
 	Put(ctx context.Context, nodes []*pb.Node, rs eestream.RedundancyStrategy,
-		pieceID client.PieceID, data io.Reader, expiration time.Time, pba *pb.PayerBandwidthAllocation, authorization *pb.SignedMessage) (successfulNodes []*pb.Node, err error)
+		pieceID psclient.PieceID, data io.Reader, expiration time.Time, pba *pb.PayerBandwidthAllocation, authorization *pb.SignedMessage) (successfulNodes []*pb.Node, err error)
 	Get(ctx context.Context, nodes []*pb.Node, es eestream.ErasureScheme,
-		pieceID client.PieceID, size int64, pba *pb.PayerBandwidthAllocation, authorization *pb.SignedMessage) (ranger.Ranger, error)
-	Delete(ctx context.Context, nodes []*pb.Node, pieceID client.PieceID, authorization *pb.SignedMessage) error
+		pieceID psclient.PieceID, size int64, pba *pb.PayerBandwidthAllocation, authorization *pb.SignedMessage) (ranger.Ranger, error)
+	Delete(ctx context.Context, nodes []*pb.Node, pieceID psclient.PieceID, authorization *pb.SignedMessage) error
 }
 
-type dialer interface {
-	dial(ctx context.Context, node *pb.Node) (ps client.PSClient, err error)
-}
+type psClientFunc func(context.Context, transport.Client, *pb.Node, int) (psclient.Client, error)
+type psClientHelper func(context.Context, *pb.Node) (psclient.Client, error)
 
-type defaultDialer struct {
-	transport transport.Client
-	identity  *provider.FullIdentity
+type ecClient struct {
+	transport       transport.Client
+	maxBuffMem      int
+	newPSClientFunc psClientFunc
 }
 
-func (dialer *defaultDialer) dial(ctx context.Context, storageNode *pb.Node) (ps client.PSClient, err error) {
-	defer mon.Task()(&ctx)(&err)
-
-	conn, err := dialer.transport.DialNode(ctx, storageNode)
-	if err != nil {
-		return nil, err
+// NewClient from the given identity and max buffer memory
+func NewClient(identity *provider.FullIdentity, mbm int) Client {
+	tc := transport.NewClient(identity)
+	return &ecClient{
+		transport:       tc,
+		maxBuffMem:      mbm,
```

Comment:
If we changing the field name in the struct then lets change the `mbm` param in the constructor to `maxBufMem` too. Also, the short for "buffer" is usually "buf", not "buff".

Reply:
`maxBufferSize`

Reply:
This value is not just about a single buffer, it's the max memory that all piece buffers can allocate together per single upload or download operation.

Reply:
`memoryLimit` then?

Reply:
`memoryLimit` sounds good.